packages/realm-react/src/useQuery.tsx

```diff
@@ -132,19 +142,47 @@ export function createUseQuery(useRealm: () => Realm): UseQueryHook {
   }
 
   return function useQueryOverload<T extends AnyRealmObject>(
-    typeOrOptions: QueryHookOptions<T> | QueryHookClassBasedOptions<T> | string | RealmClassType<T>,
+    typeOrOptionsOrQuery:
+      | QueryHookOptions<T>
+      | QueryHookClassBasedOptions<T>
+      | string
+      | RealmClassType<T>
+      | QueryCallback<T>,
     queryOrDeps: DependencyList | QueryCallback<T> = identity,
-    deps: DependencyList = [],
+    depsOrPartialOptions: DependencyList | QuaryHookPartialOptions<T> = [],
   ): Realm.Results<T> {
-    if (typeof typeOrOptions === "string" && typeof queryOrDeps === "function") {
+    if (
+      typeof typeOrOptionsOrQuery === "string" &&
+      typeof queryOrDeps === "function" &&
+      Array.isArray(depsOrPartialOptions)
+    ) {
+      /* eslint-disable-next-line react-hooks/rules-of-hooks -- We're calling `useQuery` once in any of the brances */
+      return useQuery({ type: typeOrOptionsOrQuery, query: queryOrDeps }, depsOrPartialOptions);
+    } else if (
+      isClassModelConstructor(typeOrOptionsOrQuery) &&
+      typeof queryOrDeps === "function" &&
+      Array.isArray(depsOrPartialOptions)
+    ) {
       /* eslint-disable-next-line react-hooks/rules-of-hooks -- We're calling `useQuery` once in any of the brances */
-      return useQuery({ type: typeOrOptions, query: queryOrDeps }, deps);
-    } else if (isClassModelConstructor(typeOrOptions) && typeof queryOrDeps === "function") {
+      return useQuery({ type: typeOrOptionsOrQuery as RealmClassType<T>, query: queryOrDeps }, depsOrPartialOptions);
+    } else if (
+      typeof typeOrOptionsOrQuery === "object" &&
+      typeOrOptionsOrQuery !== null &&
+      Array.isArray(depsOrPartialOptions)
+    ) {
       /* eslint-disable-next-line react-hooks/rules-of-hooks -- We're calling `useQuery` once in any of the brances */
-      return useQuery({ type: typeOrOptions as RealmClassType<T>, query: queryOrDeps }, deps);
-    } else if (typeof typeOrOptions === "object" && typeOrOptions !== null) {
+      return useQuery(typeOrOptionsOrQuery, Array.isArray(queryOrDeps) ? queryOrDeps : depsOrPartialOptions);
+    } else if (
+      typeof typeOrOptionsOrQuery === "function" &&
+      Array.isArray(queryOrDeps) &&
+      typeof depsOrPartialOptions === "object" &&
+      depsOrPartialOptions !== null
+    ) {
       /* eslint-disable-next-line react-hooks/rules-of-hooks -- We're calling `useQuery` once in any of the brances */
-      return useQuery(typeOrOptions, Array.isArray(queryOrDeps) ? queryOrDeps : deps);
+      return useQuery(
+        { ...(depsOrPartialOptions as QuaryHookPartialOptions<T>), query: typeOrOptionsOrQuery as QueryCallback<T> },
```

Comment:
Would it be possible to write this using a type assert function instead? I.e. something which test the value and return `: assert arg is QueryHookPartialOptions<T>` instead of using the `as QueryHookPartialOptions<T>`?